src/acftbias/GsiAircraftBiasConverter.cc

```diff
@@ -0,0 +1,109 @@
+#include "eckit/filesystem/PathName.h"
+
+#include "ioda/ObsGroup.h"
+#include "ioda/Engines/HH.h"
+
+#include "oops/util/missingValues.h"
+
+#include "GsiAircraftBiasReader.h"
+
+ioda::ObsGroup makeObsBiasObject(ioda::Group &empty_base_object,
+                                 const std::string & coeffile,
+                                 const std::vector<std::string> & tailIds,
+                                 const std::vector<std::string> & predictors) {
+  /// Predictors
+  long numPreds = predictors.size();
+  long numIds = tailIds.size();
+
+  /// Creating dimensions: n
+  ioda::NewDimensionScales_t newDims {
+      ioda::NewDimensionScale<int>("nvars", 1),
+      ioda::NewDimensionScale<int>("nrecs", numIds)
+  };
+  
+  /// Construct an ObsGroup object, with 2 dimensions npred, nobs
+  ioda::ObsGroup ogrp = ioda::ObsGroup::generate(empty_base_object, newDims);
+  
+  /// Create tail IDs and predictors variable
+  ioda::Variable tailIdsVar = ogrp.vars.createWithScales<std::string>("tail_ids",
+                                    {ogrp.vars["nrecs"]});
+  tailIdsVar.write(tailIds);
+    
+  /// Create 2D bias coefficient variable
+  Eigen::ArrayXXf biascoeffs(numIds, numPreds);
+
+  readObsBiasCoefficients(coeffile, biascoeffs); 
+
+  for (int i = 0; i < numPreds; ++i) {
+    /// Set up the creation parameters for the bias coefficients variable
+    ioda::VariableCreationParameters float_params;
+    float_params.chunk = true;               // allow chunking
+    float_params.compressWithGZIP();         // compress using gzip
+    float missing_value = util::missingValue(missing_value);
+    float_params.setFillValue<float>(missing_value);
+    
+    // Access first predictor and
+    Eigen::ArrayXf subVar = biascoeffs.col(i);
+      
+    // Create a variable for bias coefficients, save bias coeffs to the variable
+    ioda::Variable biasVar = ogrp.vars.createWithScales<float>("biasCoefficients/"+predictors[i],
+                       {ogrp.vars["nvars"], ogrp.vars["nrecs"]}, float_params);
+    biasVar.writeWithEigenRegular(subVar);
+  }
+
+    
+  return ogrp;
+}
+
+
+int main(int argc, char** argv) {
+  /// Open yaml with configuration for this converter
+  ASSERT(argc >= 2);
+  eckit::PathName configfile = argv[1];
+  eckit::YAMLConfiguration config(configfile);
+  
+  /// Grab input coeff file
+  const std::string coeffile = config.getString("input coeff file");
+
+  /// Use function to grab tail IDs from input coeff file
+  const std::vector<std::string> tailIds = findTailIds(coeffile);
+
+  /// Read from config file "output"
+  std::vector<eckit::LocalConfiguration> configs = config.getSubConfigurations("output");
+
+  const std::string output_filename = configs[0].getString("output file");
+  const std::vector<std::string> predictors = configs[0].getStringVector("predictors");
+  
+  /// I am defining this because the other code is not and I dont know where it came from
+  const int gsi_npredictors = 9;
```

Comment:
this should be 3, 3 predictors, each has coeffs, ob errors and background errors, right?